ast/desugar/Desugar.cc

```diff
@@ -1220,7 +1235,16 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
                 } else if (isa_reference(recv)) {
                     auto cond = MK::cpRef(recv);
                     auto body = MK::cpRef(recv);
-                    auto elsep = MK::Assign(loc, std::move(recv), std::move(arg));
+                    ExpressionPtr elsep;
+                    ast::Send *tlet;
+                    if (recvIsIvarLhs && (tlet = asTLet(arg))) {
+                        auto val = std::move(tlet->getPosArg(0));
+                        tlet->getPosArg(0) = MK::cpRef(recv);
+                        auto decl = MK::Assign(loc, MK::cpRef(recv), std::move(arg));
+                        elsep = MK::InsSeq1(loc, std::move(decl), MK::Assign(loc, std::move(recv), std::move(val)));
+                    } else {
+                        elsep = MK::Assign(loc, std::move(recv), std::move(arg));
+                    }
```

Comment:
Non-blocking: a comment showing the shape of the expression here would be nice.

Alternatively, cleaning up this code such that the left-hand side is not called `recv` (and maybe the right-hand side is not called `arg`?) might make it sufficiently clear without the diagram

Reply:
I renamed `recv` to `lhs`.